test/test_traversal.py

```diff
@@ -0,0 +1,210 @@
+            # These are expected nodes for a given depth
+            "depths_results": {
+                0: [root_node],
+                1: [root_node, foo1, foo2, foo3, foo4, test1],
+                2: [root_node, foo1, foo2, foo3, foo4, test1, test2, test3, test4, test6, test7],
+                3: [
+                    root_node,
+                    foo1,
+                    foo2,
+                    foo3,
+                    foo4,
+                    test1,
+                    test2,
+                    test3,
+                    test4,
+                    test5,
+                    test6,
+                    test7,
+                    test8
+                ],
+            },
+        }
+
+        return nodes
+
+
+@pytest.mark.parametrize("mode", ("bfs", "dfs"))
+def test_default_traversal(fake_nodes, fake_graph, mode):
+    """
+    Default traversal should return all nodes
+    """
+    with fake_graph.session_scope():
+        root = fake_graph.nodes(models.FooBar).first()
+        traversal = [n.node_id for n in root.traverse(mode=mode)]
+
+        nodes_all_set = [n.node_id for n in fake_graph.nodes().all()]
+
+    assert len(traversal) == len(nodes_all_set)
+    assert set(traversal) == set(nodes_all_set)
+
+
+@pytest.mark.parametrize("mode", ("bfs", "dfs"))
+def test_default_traversal(fake_nodes, fake_graph, mode):
+    """
+    Traversal with predicate should return only self.sysan_flag_nodes
+    """
+    with fake_graph.session_scope():
+        root = fake_graph.nodes(models.FooBar).first()
+
+        gen = root.traverse(mode=mode, edge_predicate=no_allowed_2_please)
+        traversal = [n.node_id for n in gen]
+
+    expected_ids = [n.node_id for n in fake_nodes["sysan_flag_nodes"]]
+
+    assert len(traversal) == len(expected_ids)
+    assert set(traversal) == set(expected_ids)
+
+
+@pytest.mark.parametrize("depth", [0, 1, 2, 3])
+@pytest.mark.parametrize("mode", ("bfs", "dfs"))
+def test_traversal_with_max_depth(depth, fake_graph, fake_nodes, mode):
+    """
+    Traversal should return only self.depths_results[depth] nodes
+    """
+    with fake_graph.session_scope():
+        root = fake_graph.nodes(models.FooBar).first()
+
+        gen = root.traverse(mode=mode, max_depth=depth)
+        traversal = [n for n in gen]
+
+    expected_ids = {n.node_id for n in fake_nodes["depths_results"][depth]}
+    traversal_ids = {n.node_id for n in traversal}
+    # make sure that traversal size is as expected
+    assert len(traversal) == len(fake_nodes["depths_results"][depth])
+    # make sure the results of the traversal are as expected
+    assert traversal_ids == expected_ids
+
+
+@pytest.mark.parametrize("mode", ("bfs", "dfs"))
+@pytest.mark.parametrize(
+    "key,expected",
+    (
+        ("test5", ["test5", "test2", "foo1", "root"]),
+        ("test3", ["test3", "foo2", "root"]),
+    ),
+)
+def test_default_traversal(fake_nodes, fake_graph, mode, key, expected):
```

Comment:
You have two other tests named `test_default_traversal`, python is going to only execute one, you probably want to rename them.